frontend/lib/resolution/return-type-inference.cpp

```diff
@@ -1299,9 +1299,14 @@ static bool helpComputeReturnType(ResolutionContext* rc,
     }
 
     // if there are no returns with a value, use void return type
-    if (fn->linkage() == Decl::EXTERN && fn->returnType() == nullptr) {
-      result = QualifiedType(QualifiedType::CONST_VAR, VoidType::get(context));
-      return true;
+    if (fn->linkage() == Decl::EXTERN) {
+      if (fn->returnType() == nullptr) {
+        result = QualifiedType(QualifiedType::CONST_VAR, VoidType::get(context));
+        return true;
+      } else {
+        result = QualifiedType(QualifiedType::CONST_VAR, ErroneousType::get(context));
```

Comment:
I think we're trying to move to using `ErroneousType` only when an error would be emitted. Since there's no error, I think `UnknownType` is what should be returned here, _or_ an error issued.